### DISCUSSION TITLE
how to handle errors in useFetch, and correctly throw errors in server routes.

### DISCUSSION THREAD

**Discussion by LoganTann** [nuxt/nuxt#17499] [Questions]
Hello, I always used Nuxt2, and it's my first time with Nuxt3

I really love the server-routes features and the ecosystem behind h3 and Nitro. Most people would separate the front-end and the back-end in two different servers, but having a very simple API while taking benefits from the SSR really suits my needs !

However, **I still haven't figured how to properly handle errors in useFetch**. Doing a console.log of data and error produces weird things in the firefox console.

Also, **what is the best way in the server routes to send a 404 error** ?

Currently, I'm just throwing an error, but it produces 500 errors

**Comment by wobsoriano** [comment]
https://github.com/nuxt/framework/discussions/5236#discussioncomment-2873301

**Reply by DesertCookie** [reply]
The link provided is dead. Would you happen to have a newer one?
